pkg/kapp/app/recorded_app.go

```diff
@@ -196,27 +191,109 @@ func (a *RecordedApp) createOrUpdate(c *corev1.ConfigMap, labels map[string]stri
 				return fmt.Errorf("Getting app: %s", err)
 			}
 
-			err = a.mergeAppUpdates(existingConfigMap, labels)
-			if err != nil {
-				return err
+			return a.updateApp(existingConfigMap, labels)
+		}
+
+		return fmt.Errorf("Creating app: %s", err)
+	}
+
+	return nil
+}
+
+func (a *RecordedApp) updateApp(existingConfigMap *corev1.ConfigMap, labels map[string]string) error {
+	err := a.mergeAppUpdates(existingConfigMap, labels)
+	if err != nil {
+		return err
+	}
+
+	_, err = a.coreClient.CoreV1().ConfigMaps(a.nsName).Update(context.TODO(), existingConfigMap, metav1.UpdateOptions{})
+	if err != nil {
+		return fmt.Errorf("Updating app: %s", err)
+	}
+
+	return nil
+}
+
+func (a *RecordedApp) RenamePrevApp(prevAppName string, labels map[string]string) error {
+	defer a.logger.DebugFunc("RenamePrevApp").Finish()
+	var c *corev1.ConfigMap
+	var err error
+
+	newConf := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      a.name,
+			Namespace: a.nsName,
+			Labels: map[string]string{
+				KappIsAppLabelKey: kappIsAppLabelValue,
+			},
+		},
+		Data: Meta{
+			LabelKey:   kappAppLabelKey,
+			LabelValue: fmt.Sprintf("%d", time.Now().UTC().UnixNano()),
+			UsedGKs:    &[]schema.GroupKind{},
+		}.AsData(),
+	}
+
+	if a.isMigrationEnabled() {
+		c, err = a.coreClient.CoreV1().ConfigMaps(a.nsName).Get(context.TODO(), a.fqName, metav1.GetOptions{})
+		if err == nil {
+			return a.updateApp(c, labels)
```

Comment:
Should we also check here the configMap we got is having a kapp app label `kapp.k14s.io/is-app` or not using `isKappApp()` method before updating it ? To make sure this is the configmap created by kapp only. 

As of now looks like we do this in `CreateOrUpdate()` method.

Reply:
Can we use the same `a.updateApp(c, labels)` even in case of `CreateOrUpdate` function . https://github.com/vmware-tanzu/carvel-kapp/blob/develop/pkg/kapp/app/recorded_app.go#L142-L147